python/lsst/obs/subaru/isr.py

```diff
@@ -295,7 +295,37 @@ def runDataRef(self, sensorRef):
         if self.config.doCrosstalk:
             self.crosstalk.run(ccdExposure)
 
+        if self.config.doWidenSaturationTrails:
+            self.widenSaturationTrails(ccdExposure.getMaskedImage().getMask())
+
+        interpolationDone = False
+
         if self.config.doBrighterFatter:
+
+            # Temporarily apply flat and dark so we can interpolate prior to B-F correction
+            if self.config.doDark:
+                darkExposure = self.getIsrExposure(sensorRef, "dark")
+                self.darkCorrection(ccdExposure, darkExposure)
+            if self.config.doFlat:
+                flatExposure = self.getIsrExposure(sensorRef, "flat")
+                self.flatCorrection(ccdExposure, flatExposure)
+
+            # Interpolate defects, saturation, and NaNs
+            if self.config.doDefect:
+                self.maskAndInterpDefect(ccdExposure, defects)
+            if self.config.doSaturation:
+                self.saturationInterpolation(ccdExposure)
+            self.maskAndInterpNan(ccdExposure)
+            interpolationDone = True
+
+            # Undo flat and dark application
```

Comment:
Doing and then undoing suggests using a context manager to simplify the main code path.